docs/bookstore/page-2/sentiment-analysis-service-for-bookstore-reviews.md

```diff
@@ -0,0 +1,414 @@
+    ```
+    
+    In this case, we will use the s2i build by adding the flag `-b=s2i`, and `-v` to see the verbose output.
+    
+    ```
+    func build -b=s2i -v
+    ```
+    
+    When the build is complete, you will see the following output:
+    
+    ```
+    🙌 Function built: <Your container registry username>/sentiment-analysis-app:latest
+    ```
+    
+    This command will build the function and push the image to the container registry. After the build is complete, you can run the function using the following command:
+    
+    ---
+    
+    **Troubleshooting**
+    
+    `❗Error: '/home/Kuack/Documents/knative/docs/code-samples' does not contain an initialized function`
+    
+    **Solution: You may want to check whether you are in the correct directory. You can use the following command to check the current directory. If you are in the right directory, and the error still occurs, try to check your `func.yaml`, as it has to contain the field `created` and the right timestamp to be treated as a valid Knative Function.**
+    
+    ---
+    
+    ```
+    func run -b=s2i -v
+    ```
+    
+    In the future, you can skip the step of `func build`, because `func run` will automatically build the function for you.
+    
+    You will see the following output if the function is running successfully:
+    
+    ```
+    ❗function up-to-date. Force rebuild with --build
+    Running
+    
+     on host port 8080
+    ---> Running application from script (app.sh) ...
+    ```
+    
+    Knative Function has an easy way to simulate the CloudEvent, you can use the following command to simulate the CloudEvent and test your function out:
+    
+    ```
+    func invoke -f=cloudevent --data='{"reviewText": "I love Knative so much"}' --content-type=application/json --type="new-review-comment" -v
+    ```
+    
+    where the `-f` flag indicates the type of the data, is either `HTTP` or `cloudevent`, and the `--data` flag is the input text. You can read more about `func invoke` [here](https://github.com/knative/func/blob/main/docs/reference/func_invoke.md){:target="_blank"}.
+    
+    In this case, you will get the full CloudEvent response:
+    
+    ```
+    Context Attributes,
+      specversion: 1.0
+      type: new-review-comment
+      source: book-review-broker
+      id: ebbcd761-3a78-4c44-92e3-de575d1f2d38
+      time: 2024-05-27T04:44:07.549303Z
+      datacontenttype: application/json
+    Extensions,
+      badwordfilter: good
+    Data,
+      {
+        "reviewText": "I love Knative so much",
+        "badWordResult": "",
+         "sentimentResult": "positive"
+      }
+    ```
+    
+### **Step 6: Deploy the function to the cluster**
+
+![Image10](images/image10.png)
+
+!!! note
+    Please enter `/sentiment-analysis-app` when you are executing the following commands.
+
+After you have finished the code, you can deploy the function to the cluster using the following command:
+
+```bash
+func deploy -b=s2i -v
```

Comment:
➜  sentiment-analysis-app git:(10b0ae63) ✗ func deploy
function up-to-date. Force rebuild with --build
Pushing function image to the registry "index.docker.io" using the "matzew" user credentials
⬆️  Deploying function to the cluster
^C
Service output:
Traceback (most recent call last):
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 29, in <module>
    main()
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 20, in main
    app = server.create(server.load(sys.argv[1]))
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/server.py", line 16, in load
    import func
  File "/workspace/func.py", line 4, in <module>
    from textblob import TextBlob
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/__init__.py", line 1, in <module>
    from .blob import Blobber, Sentence, TextBlob, Word, WordList
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/blob.py", line 26, in <module>
    import nltk
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/__init__.py", line 153, in <module>
    from nltk.translate import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/__init__.py", line 24, in <module>
    from nltk.translate.meteor_score import meteor_score as meteor
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/meteor_score.py", line 13, in <module>
    from nltk.corpus import WordNetCorpusReader, wordnet
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/__init__.py", line 64, in <module>
    from nltk.corpus.reader import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/__init__.py", line 106, in <module>
    from nltk.corpus.reader.panlex_lite import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/panlex_lite.py", line 15, in <module>
    import sqlite3
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/__init__.py", line 57, in <module>
    from sqlite3.dbapi2 import *
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/dbapi2.py", line 27, in <module>
    from _sqlite3 import *
ImportError: libsqlite3.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 29, in <module>
    main()
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 20, in main
    app = server.create(server.load(sys.argv[1]))
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/server.py", line 16, in load
    import func
  File "/workspace/func.py", line 4, in <module>
    from textblob import TextBlob
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/__init__.py", line 1, in <module>
    from .blob import Blobber, Sentence, TextBlob, Word, WordList
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/blob.py", line 26, in <module>
    import nltk
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/__init__.py", line 153, in <module>
    from nltk.translate import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/__init__.py", line 24, in <module>
    from nltk.translate.meteor_score import meteor_score as meteor
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/meteor_score.py", line 13, in <module>
    from nltk.corpus import WordNetCorpusReader, wordnet
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/__init__.py", line 64, in <module>
    from nltk.corpus.reader import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/__init__.py", line 106, in <module>
    from nltk.corpus.reader.panlex_lite import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/panlex_lite.py", line 15, in <module>
    import sqlite3
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/__init__.py", line 57, in <module>
    from sqlite3.dbapi2 import *
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/dbapi2.py", line 27, in <module>
    from _sqlite3 import *
ImportError: libsqlite3.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 29, in <module>
    main()
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 20, in main
    app = server.create(server.load(sys.argv[1]))
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/server.py", line 16, in load
    import func
  File "/workspace/func.py", line 4, in <module>
    from textblob import TextBlob
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/__init__.py", line 1, in <module>
    from .blob import Blobber, Sentence, TextBlob, Word, WordList
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/blob.py", line 26, in <module>
    import nltk
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/__init__.py", line 153, in <module>
    from nltk.translate import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/__init__.py", line 24, in <module>
    from nltk.translate.meteor_score import meteor_score as meteor
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/meteor_score.py", line 13, in <module>
    from nltk.corpus import WordNetCorpusReader, wordnet
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/__init__.py", line 64, in <module>
    from nltk.corpus.reader import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/__init__.py", line 106, in <module>
    from nltk.corpus.reader.panlex_lite import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/panlex_lite.py", line 15, in <module>
    import sqlite3
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/__init__.py", line 57, in <module>
    from sqlite3.dbapi2 import *
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/dbapi2.py", line 27, in <module>
    from _sqlite3 import *
ImportError: libsqlite3.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 29, in <module>
    main()
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 20, in main
    app = server.create(server.load(sys.argv[1]))
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/server.py", line 16, in load
    import func
  File "/workspace/func.py", line 4, in <module>
    from textblob import TextBlob
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/__init__.py", line 1, in <module>
    from .blob import Blobber, Sentence, TextBlob, Word, WordList
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/blob.py", line 26, in <module>
    import nltk
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/__init__.py", line 153, in <module>
    from nltk.translate import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/__init__.py", line 24, in <module>
    from nltk.translate.meteor_score import meteor_score as meteor
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/meteor_score.py", line 13, in <module>
    from nltk.corpus import WordNetCorpusReader, wordnet
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/__init__.py", line 64, in <module>
    from nltk.corpus.reader import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/__init__.py", line 106, in <module>
    from nltk.corpus.reader.panlex_lite import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/panlex_lite.py", line 15, in <module>
    import sqlite3
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/__init__.py", line 57, in <module>
    from sqlite3.dbapi2 import *
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/dbapi2.py", line 27, in <module>
    from _sqlite3 import *
ImportError: libsqlite3.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 29, in <module>
    main()
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 20, in main
    app = server.create(server.load(sys.argv[1]))
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/server.py", line 16, in load
    import func
  File "/workspace/func.py", line 4, in <module>
    from textblob import TextBlob
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/__init__.py", line 1, in <module>
    from .blob import Blobber, Sentence, TextBlob, Word, WordList
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/blob.py", line 26, in <module>
    import nltk
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/__init__.py", line 153, in <module>
    from nltk.translate import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/__init__.py", line 24, in <module>
    from nltk.translate.meteor_score import meteor_score as meteor
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/meteor_score.py", line 13, in <module>
    from nltk.corpus import WordNetCorpusReader, wordnet
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/__init__.py", line 64, in <module>
    from nltk.corpus.reader import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/__init__.py", line 106, in <module>
    from nltk.corpus.reader.panlex_lite import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/panlex_lite.py", line 15, in <module>
    import sqlite3
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/__init__.py", line 57, in <module>
    from sqlite3.dbapi2 import *
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/dbapi2.py", line 27, in <module>
    from _sqlite3 import *
ImportError: libsqlite3.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 29, in <module>
    main()
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 20, in main
    app = server.create(server.load(sys.argv[1]))
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/server.py", line 16, in load
    import func
  File "/workspace/func.py", line 4, in <module>
    from textblob import TextBlob
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/__init__.py", line 1, in <module>
    from .blob import Blobber, Sentence, TextBlob, Word, WordList
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/blob.py", line 26, in <module>
    import nltk
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/__init__.py", line 153, in <module>
    from nltk.translate import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/__init__.py", line 24, in <module>
    from nltk.translate.meteor_score import meteor_score as meteor
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/meteor_score.py", line 13, in <module>
    from nltk.corpus import WordNetCorpusReader, wordnet
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/__init__.py", line 64, in <module>
    from nltk.corpus.reader import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/__init__.py", line 106, in <module>
    from nltk.corpus.reader.panlex_lite import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/panlex_lite.py", line 15, in <module>
    import sqlite3
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/__init__.py", line 57, in <module>
    from sqlite3.dbapi2 import *
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/dbapi2.py", line 27, in <module>
    from _sqlite3 import *
ImportError: libsqlite3.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 29, in <module>
    main()
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/__main__.py", line 20, in main
    app = server.create(server.load(sys.argv[1]))
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/parliament/server.py", line 16, in load
    import func
  File "/workspace/func.py", line 4, in <module>
    from textblob import TextBlob
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/__init__.py", line 1, in <module>
    from .blob import Blobber, Sentence, TextBlob, Word, WordList
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/textblob/blob.py", line 26, in <module>
    import nltk
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/__init__.py", line 153, in <module>
    from nltk.translate import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/__init__.py", line 24, in <module>
    from nltk.translate.meteor_score import meteor_score as meteor
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/translate/meteor_score.py", line 13, in <module>
    from nltk.corpus import WordNetCorpusReader, wordnet
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/__init__.py", line 64, in <module>
    from nltk.corpus.reader import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/__init__.py", line 106, in <module>
    from nltk.corpus.reader.panlex_lite import *
  File "/layers/paketo-buildpacks_pip-install/packages/lib/python3.10/site-packages/nltk/corpus/reader/panlex_lite.py", line 15, in <module>
    import sqlite3
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/__init__.py", line 57, in <module>
    from sqlite3.dbapi2 import *
  File "/layers/paketo-buildpacks_cpython/cpython/lib/python3.10/sqlite3/dbapi2.py", line 27, in <module>
    from _sqlite3 import *
ImportError: libsqlite3.so.0: cannot open shared object file: No such file or directory

deploy error: knative deployer failed to wait for the Knative Service to become ready: context canceled
Error: knative deployer failed to wait for the Knative Service to become ready: context canceled

Reply:
Please say that `s2i` is required and the default `pack` does NOT work